### PR TITLE
Start migration from cbindgen to bindgen

### PR DESCRIPTION
**Pull request by DevOrc**
Fix for #78 - has rust types generated by the header instead of vice-versa. Currently still needs a lot of work (most of the original codebase is deleted 😞 ) but I am slowly reimplementing everything with the generated types. Currently the triangle and compute examples _should_ work on Windows and Linux. Anyways, thought I would post this here to get initial thoughts before I finished adding everything. 

## Big changes
- all of the header types are generated at build time and inserted into the `crate::native` module
- enums are mapped from their c types to wgt using a macro (map_enum!)
- two c headers (`ffi/webgpu.h` for things in the [webgpu-native](https://github.com/webgpu-native/webgpu-headers) headers, and `ffi/wgpu.h` for stuff unique to wgpu-native)

## Small Notes
- auto-generated bindings are modified at build time to change all of the *Impl to Ids (`BindGroupImpl` -> `wgc::id::BindGroupId`)
- libclang now required to build (required by bindgen)
- enum constants are i32 on windows and are u32 on macos/linux
- example shaders are now written in WGSL
- Had to change some of the makefiles for windows c development (if anyone has a better way to write makefiles please let me know :smile:)

## Stuff that still needs to be added
- [x] MacOS Surfaces
- [ ] Wayland Surfaces
- [x]  most commands
- [x] undelete / update the capture example
- [x] logging
- [x] Textures
- [x] Fix IOS
- [ ] Include webgpu.h in release CI

### CODE REVIEW

File: ffi/webgpu.h [hunk 1]
```diff
@@ -0,0 +1,1208 @@
+// BSD 3-Clause License
```
Comment (kvark) [inline ffi/webgpu.h:1]: if this is taken from webgpu-headers, can we set this up as a git submodule?

**Comment by kvark** [comment]
@DevOrc this is absolutely critical work, even if it's very routine. I don't know how to express the gratitude to you for doing this! 🥇

**Comment by kvark** [comment]
Is this ready to go?

**Comment by DevOrc** [comment]
Almost - need to finish porting some of the functions. Will probably be ready by tonight or tomorrow.

**Review by kvark — APPROVED** [review]
bors r+

**Pull request merged**